fix: add warn log when a file failed to be ingested

enzo-cappa opened the pull request:
Add a WARN level log when a file failed to be ingested according to the copy history.

sfc-gh-tzhang committed:
NO-SNOW: match pom_confluent.xml with pom.xml (#531)

match pom_confluent.xml with pom.xml

sfc-gh-wfateem committed:
[SNOW-726924] Add New jvm.nonProxy.hosts Parameter and update JDBC to 3.13.23 (#533)

The jvm.nonProxy.hosts parameter can be added to the connector's configuration
file and is the equivalent to the JVM argument -Dhttp.nonProxyHosts. If a user
provides that parameter and the connector runs in a JVM where -Dhttp.nonProxyHosts
was set, then we will merge the two lists of hosts. In the future, we'll want another
method of adding proxy configurations without resorting to System.setProperty() calls

sfc-gh-rcheng committed:
[SNOW-692657] Create one client per connector instead of one client per task (#528)

* ut pass

* oops i think i broke master

* one test passes

* one test fixed

* thinking about just making the client directly

* tests pass but static

* provider passes

* fix task it

* its late but it doesnt all pass. private key schema issues

* format to save my own reviews

* just one test ah

* that was a dumb bug lol

* tests pass

* format

* get ingestsdkprovider on task.start not task(). sorry for got to push this earlier

* 100% code coverage yay + nits

* i think i see a betterfix?

* ok thats cleaner, tests running but its 8pm and cats are clingy

* tmp save

* public singleton

* running tests, format

* testutils

* format

* thinking about inheritance?

* framework for clientmanager?

* another layer

* javadocs and format

* rename and move test, remove unneeded providers

* format

* small nits

* added retry

* map created, tmp save

* remove retrycount

* impl, workign on tests

* overnight save

* 100 on clientmanager

* save... think im gonna move the layers around a bit

* well everything builds... if i comment out the tests lol

* sunny kcclient tests

* format

* fixing tests...

* tp passes with some verification issues

* well tp passes... and i did probably too many changes

* format

* it pass, cc 100 on manager

* 100 cc on cm and kcclient, tests should pass - running now

* nits

* clientmanager prepend streaming

* uncommented and simplified it, format

* nits, make it more simple

* nits

* comment changes

* comment for taskid dep

* added it and nits

* format

* clientid

* typos are the bane of my expectation

* create client loop cleanup

* i really should write a script to format for meeee

* remove comment

* static get and set

* remove todos

* method streaming

sfc-gh-rcheng committed:
Fix client off by one issue (#540)

* wrong iterations

* fix it

* comments and todo

* double trouble is truely trouble

* multi logs work, still need client mapping test

* format

sfc-gh-rcheng committed:
Refactor sink task integration tests to be more readable (#543)

* wrong iterations

* fix it

* comments and todo

* double trouble is truely trouble

* multi logs work, still need client mapping test

* all pass

* format and quick comment

* tests pass, need reset method

* no sunny days in seattle

* move to diff file

* formatttttt

* ignore logging test bc jdbc flakiness

snyk-bot committed:
connect-api Bump to 3.4.0 fix: pom.xml to reduce vulnerabilities (#539)

Add to pom_confluent

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEKAFKA-3317161

sfc-gh-japatel committed:
Modify pom.xml to use jdbc defined in connector and not from ingest sdk (#546)

* Modify pom.xml to use jdbc defined in connector and not from ingest sdk

* Fail fast to false to not interuppt other workflows

* Fix incorrect format for data as of new version

sfc-gh-japatel committed:
SNOW-740327 Create config for using Arrow BDEC file format + Bump Client SDK 1.0.3-beta (#541)

* SNOW-740327 Create config for using Arrow BDEC file format

- Also bumps up the latest IngestSDK - 1.0.3-beta

* Addressed Revi's comments

* Google jfmt

* Remove unnecessary layer for file type enum

- Accept the bdec version directly. We will only support ONE and THREE

* Remove validator and default values, send everything to Ingest SDK

* Fix comments from revi's code and integrate the reset method

* Google jfmt

* Ingest SDK expects a number and not a string numbered

* Set default to empty string

* Description change

sfc-gh-japatel committed:
no-snow Add connector name in KC client name for better debugging the client … (#547)

* Add connector name in KC client name for better debugging the client history view

* Google jfmt

enzo-cappa committed:
fix: add warn log when a file failed to be ingested

Add a WARN level log when a file failed to be ingested according to the copy history.

migruealt commented on src/main/java/com/snowflake/kafka/connector/internal/SnowflakeIngestionServiceV1.java:
```diff
@@ -144,7 +141,14 @@ public Map<String, InternalUtils.IngestedFileStatus> readIngestReport(final List
+            LOG_INFO_MSG("searched {} files in ingest report, found {}", files.size(), numOfRecords);
+            // Log errors
+            if (InternalUtils.IngestedFileStatus.FAILED.equals(ingestionStatus) || InternalUtils.IngestedFileStatus.PARTIALLY_LOADED.equals(ingestionStatus)) {
+              LOG_WARN_MSG("Failed to load file {}\n\tError: {}\n\tColumn: {}\n\tChar pos: {}",
```
should this be a warning?

migruealt reviewed (approved):
👏

sfc-gh-japatel committed:
Bump Ingest SDK to 1.1.0 and release 1.9.0 (#552)

* Bump Ingest SDK and release 1.9.0

* Debug failing test

* Try simple fix

* revert debugging failed test

sfc-gh-japatel committed:
NO-SNOW Revert one client change, JDBC upgrade and config for arrow and parquet (#555)

* Revert "no-snow Add connector name in KC client name for better debugging the client … (#547)"

This reverts commit f7df64df6a3288664d38f4f2488e4abbb994e669.

* Revert "Refactor sink task integration tests to be more readable (#543)"

This reverts commit 1b24187c864322549e60963db58bc75bb8ca3924.

* Ignore flaky test

* Revert "Fix client off by one issue (#540)"

This reverts commit a5e9b456b940b759cf8f6d30f1323107571fa7bf.

* Revert "[SNOW-692657] Create one client per connector instead of one client per task (#528)"

This reverts commit 0b4b1cb234ee94dc3627555f63a140c723f871ee.

* Revert "SNOW-740327 Create config for using Arrow BDEC file format + Bump Client SDK 1.0.3-beta (#541)"

This reverts commit c5f8aed1a3474fe2cc46ba9fb41faec29341b3d9.

* Revert "[SNOW-726924] Add New jvm.nonProxy.hosts Parameter and update JDBC to 3.13.23 (#533)"

This reverts commit d2c35614f940f2b20fb345ecadcb3918c610f637.

* Revert "Modify pom.xml to use jdbc defined in connector and not from ingest sdk (#546)"

This reverts commit 79051a16813a34825a8454ae7d6291837b118c50.

sfc-gh-japatel committed:
SNOW-740327 Create config for using Arrow BDEC file format - Redo after revert of earlier commits (#556)

sfc-gh-tzhang committed:
NO-SNOW: allow StringConverter when schematization is in the config but set to false (#560)

* fix

* format

* add tests

sfc-gh-japatel committed:
NO-SNOW Release 1.9.1 for Snowpipe Streaming (#559)

sfc-gh-japatel committed:
NO-SNOW Include bcpkix dependency in Confluent Uber Jar so be able to not add them manually at runtime (#563)

revert major change from 696f69280e652c788ae825ed9714a79eaff1ca58

sfc-gh-japatel committed:
SNOW-758111 Use telemetry service for all exceptions wherever possible (#569)

sfc-gh-japatel committed:
[NO-SNOW] MINOR - Update README.md (#572)

sfc-gh-japatel committed:
NO-SNOW Disable Exactly Once delivery guarantee tests for Snowpipe (#577)

sfc-gh-japatel committed:
NO-SNOW Best practice: Use fail-safe plugin for IT tests (#578)

sfc-gh-japatel committed:
SNOW-760171 Use pom_confluent.xml in running confluent tests (#573)

### Benefits:
- Catch issues before releasing a zip in Confluent Hub
- Use pom_confluent.xml in our tests. We only used pom.xml until now and no way to check the diff between those two. 
- For private key which are encrypted, we have to deploy fips jar to confluen setup, this test would have caught that. 

### Details
- Rename build_apache.jar with build_runtime_jar
- Use pom_confluent.xml in Running End to end test against Confluent
- Pass in confluent argument when we want to use confluent as kafka runtime. 
- Add description in sh script. 
- pom_confluent uses https://docs.confluent.io/platform/current/connect/kafka-connect-maven-plugin/site/plugin-info.html plugin to create a zip file which gets released to Confluent Hub. 

Update: [For confluent tests]
- For all versions, we point plugin path to the zip file we extract after running build_runtime_jar.sh which runs the connect maven plugin.

sfc-gh-japatel committed:
No-SNOW Update README.md Links (#581)

sfc-gh-japatel committed:
SNOW-763169 Use maven Cache - 4X improvement in runtime of Github Actions for builds (#579)

enzo-cappa committed:
Merge branch 'master' of github.com:enzo-cappa/snowflake-kafka-connector into feat/log_failed_files

enzo-cappa committed:
fix: check nullability before using variables

Validate null values before using properties.

enzo-cappa committed:
fix: undo check on imports

sfc-gh-japatel committed:
SNOW-760171 Part 1: Test to verify DLQ results  (#585)

- Invalid Json Serialized
- Schema Mapping incorrect column value

sfc-gh-tzhang committed:
SNOW-759517, SNOW-743629, SNOW-733467: simplify offset management logic in KC (#566)

SNOW-759517, SNOW-743629, SNOW-733467

- Instead of using 5 fields for offset management, we try to simplify the code and use just one, see processedOffset. We also keep offsetPersistedInSnowflake but that's just for tracking purpose
- We always use the offset token from server side as long as it's valid, when there is no valid offset token at server side, we just trust what kafka sends us
- Add some special handling for schematization schema evolution, and add a workaround for a Ingest SDK issue that causes test flakyness
- Fix tests due to behavior changes and add more validations

sfc-gh-rcheng committed:
[SNOW-659469] List all invalid parameters  (#586)

sfc-gh-japatel committed:
SNOW-756851 Part 1: Refactor code to create telemetry service for Snowpipe Streaming (#583)

sfc-gh-rcheng committed:
Reenable accidentally disabled config tests from PR 586 (#588)

sfc-gh-rcheng committed:
One client change 1/3: add parameter to enable one client optimization (#589)

Co-authored-by: Jay Patel <jay.patel@snowflake.com>

sfc-gh-tzhang committed:
SNOW-768488: Accept any offset from kafka if InvalidOffsetException is thrown during startup (#587)

It's possible that customer stops their KC and the data stored in kafka expired, when the KC is restarted, it will use the channel's offset token and resetting the offset will result in an InvalidOffsetException, this PR handles this case and we will accept what kafka sends us instead

sfc-gh-rcheng committed:
[SNOW-748118] Remove delivery guarantee config param (#551)

sfc-gh-japatel committed:
NO-SNOW Refactor Test suites  + Disable failing tests to unblock other PRs (#598)

sfc-gh-japatel committed:
NO-SNOW Use a different github key to reduce github API limits (#600)

sfc-gh-japatel committed:
SNOW-480523 Upgrade JDBC to 3.13.25 to use downscope gcs token instea… (#593)

sfc-gh-rcheng committed:
Add cla bot from snowflake-connector-python repo (#597)

sfc-gh-japatel committed:
SNOW-774533 Enable Json & Proxy Test - Bug in setting up Confluent Tests (#605)

sfc-gh-japatel committed:
SNOW-480523 Fix breaking change in upgrade to JDBC 3.13.25 (#607)

sfc-gh-rcheng committed:
Add format script (#611)

sfc-gh-japatel committed:
SNOW-774533 Fix DLQ bug, re-enable test (#604)

sfc-gh-japatel committed:
Add telemetry for fatal errors in TopicPartitionChannel (#601)

sfc-gh-japatel committed:
NO-SNOW Add caching layer for maven to speed up Github action (#613)

sfc-gh-japatel committed:
[SNOW-726924] Add New jvm.nonProxy.hosts Parameter - Undo Revert  (#614)

Co-authored-by: Waleed Fateem <72769898+sfc-gh-wfateem@users.noreply.github.com>

sfc-gh-japatel committed:
Vuln Update: Upgrade JDBC to 3.13.29 (#616)

sfc-gh-rcheng committed:
Add resilience testing (#617)

sfc-gh-sscott committed:
SNOW-735650: sfc-jenkins -> sfc-eng-jenkins s3 bucket change (#532)

sfc-gh-japatel committed:
NO-SNOW Delete Full tests as cron jobs as they are now part of every PR (#621)

sfc-gh-tzhang committed:
NO-SNOW: fix offset token behavior when resetting the offset to data that expired (#619)

- Remove the change that catch InvalidOffsetException during reseting the offset since the suggestion of setting auto.reset.offset doesn't really work
- Add back isOffsetResetInKafka with modifications to support exactly once delivery for a special case

sfc-gh-japatel committed:
SNOW-803249 Remove confluent 5.5.11 since it is no longer supported (#623)

sfc-gh-japatel committed:
[NO-SNOW] Use latest version of Ingest SDK (#622)

sfc-gh-japatel committed:
NO-SNOW Release version 1.9.2 (#624)

sfc-gh-tzhang committed:
SNOW-791549: Infer NULL to VARCHAR instead of VARIANT since ingestion timestamp into VARIANT column is not allowed (#625)

when the column data is NULL, we don't know which data to infer, so we use VARIANT currently, but VARIANT doesn't work with timestamp using the current SDK, this PR will infer the data type to VARCHAR instead of VARIANT.

sfc-gh-rcheng committed:
Use MDC context instead of kc and logger task (#568)

sfc-gh-tzhang committed:
NO-SNOW: fix commit offset behavior for Snowpipe (#631)

The change introduced in afd3588 was not meant for Snowpipe and it's causing reset of offset to 0 during KC restart, this PR contains the change to revert back to the old behavior for Snowpipe

sfc-gh-rcheng committed:
One client change 2/3: Add client provider (#590)

sfc-gh-rcheng committed:
One client change 3/3: Call into client provider and corresponding resilience tests (#609)

sfc-gh-rcheng committed:
Enable topic regexes for table2topic mapping (#628)

sfc-gh-japatel committed:
SNOW-480523 Fix Downscope URL for GCS (#634)

enzo-cappa committed:
Merge branch 'master' of github.com:enzo-cappa/snowflake-kafka-connector into feat/log_failed_files

sfc-gh-rcheng committed:
[Buffer Flush 1/2] Refactor buffer threshold variables and methods to be more consistent (#639)

enzo-cappa committed:
fix: remove potential pii data and format

sfc-gh-rcheng committed:
Fix cla bot by using correct github token (#640)

enzo-cappa committed:
Merge pull request #2 from snowflakedb/master

Update

enzo-cappa committed:
Merge branch 'master' of github.com:enzo-cappa/snowflake-kafka-connector into feat/log_failed_files

sfc-gh-rcheng committed:
Revert kafka put and precommit log message to info (#637)

sfc-gh-rcheng committed:
Remove CLA bot and update README.md to include cla and formatter requirements (#643)

enzo-cappa committed:
Merge branch 'master' of ssh://github.com/snowflakedb/snowflake-kafka-connector into feat/log_failed_files
